### ISSUE TITLE
wrangler dev --local does a permanent redirect

### ISSUE THREAD

**Issue by aivLuk** [cloudflare/workers-sdk#3078]
### Which Cloudflare product(s) does this pertain to?

Wrangler

### What version of `Wrangler` are you using?

2.15.0

### What operating system are you using?

Mac

### Describe the Bug

Not sure if it's an issue or unclarity how actually wrangler dev local environment works, but when I define the route in `wrangler.toml` and run `wrangler dev --local` it tries to open `http://127.0.0.1` but quickly redirects with 301 to the route defined in wrangler.toml. I'm not sure how to develop locally with these redirects.

Command that I run: `wrangler dev --local --live-reload`

Simplified version of worker:
```
export async function handleEvent(event: FetchEvent): Promise<Response> {
  return fetch(event.request);
}

addEventListener('fetch', (event: FetchEvent) => {
  event.passThroughOnException();
  event.respondWith(handleEvent(event));
});
```

wrangler.toml:
```
name = "my-worker"
route = "my-worker.com/*"
main = "dist/worker.js"
compatibility_date = "2023-04-06"
node_compat = true
```

So basically running dev command with this setup and opening the browser for `http://127.0.0.1` will quickly redirect to `example.com` and I lose the ability to develop locally. 
If I remove the route from wrangler.toml I get an error:
`If you're trying to fetch from an origin server, make sure you've set the 'upstream' option.`
Then setting the `--local-upstream example.com` again redirects to `example.com`

**Comment by aivLuk** [comment]
Okay, fixed it. It seems I needed to add --local-protocal=https`

**Comment by admah** [comment]
@aivLuk I'm glad you were able to figure out the issue!
